tests/unit/keyinput/test_basekeyparser.py

```diff
@@ -90,6 +92,29 @@ def test_read_config_valid(self):
         assert 'foo' in kp.bindings
         assert 'ctrl+x' in kp.special_bindings
 
+    def test_on_keyconfig_changed(self):
+        """Test the handling of changes in the config."""
+        kp = basekeyparser.BaseKeyParser(0, supports_count=False,
+                                         supports_chains=False)
+        kp.read_config = mock.Mock()
+        kp._modename = None
```

Comment:
This is `None` by default anyways, right? Then no need to set it explicitely - maybe even do `assert kp._modename is None`